python/ray/train/tests/test_jax_gpu.py

```diff
@@ -0,0 +1,114 @@
+import pytest
+import ray
+from ray.air.config import ScalingConfig
+from ray.train.constants import TRAINING_ITERATION
+from ray.train.examples.jax_mnist_example import (
+    train_func as jax_mnist_train_func,
+)
+from ray.air import session
+from ray.train.jax import JaxTrainer
+
+import jax
+import numpy as np
+
+
+from ray import tune
+from ray.tune.tune_config import TuneConfig
+from ray.tune.tuner import Tuner
+
+
+@pytest.fixture
+def ray_start_4_cpus_2_gpus():
+    address_info = ray.init(num_cpus=4, num_gpus=2)
+    yield address_info
+    # The code after the yield will run as teardown code.
+    ray.shutdown()
+
+
+def test_jax_get_device(ray_start_4_cpus_2_gpus):
+    def _train_fn(x):
+        return jax.lax.psum(x, "i")
+
+    def train_fn():
+        """Creates a barrier across all hosts/devices."""
+        session.report(
+            dict(devices=jax.pmap(_train_fn, "i")(np.ones(jax.local_device_count())))
+        )
+
+    num_gpus_per_worker = 2
+    trainer = JaxTrainer(
+        train_fn,
+        scaling_config=ScalingConfig(
+            num_workers=1,
+            use_gpu=True,
+            resources_per_worker={"GPU": num_gpus_per_worker},
+        ),
+    )
+
+    results = trainer.fit()
+    devices = results.metrics["devices"]
+    assert len(devices) == 2 and devices[0] == 2 and devices[1] == 2
```

Comment:
would be great to add a comment explaining what's going on this test!

Seems like we have 2 devices, and each device is summing the total number of devices?